var/spack/repos/builtin/packages/graphviz/package.py

```diff
@@ -34,11 +34,13 @@ class Graphviz(AutotoolsPackage):
 
     version('2.38.0', '5b6a829b2ac94efcd5fa3c223ed6d3ae')
 
+    # Swig can be enabled on it's own, though it doesn't do much.  It
+    # will also be enabled if any of the languages are enabled.
+    variant('swig', default=False,
+            description='Enable swig language binding tools')
```

Comment:
I lifted the two lists up to the class level and thought that this might work, but I'm tripping over where to get the `spec`.

```python
    # These language bindings have been tested, we know they work.
    tested_bindings = ('+java', '+perl')

    # These language bindings have not yet been tested.  They
    # likely need additional dependencies to get working.
    untested_bindings = (
        '+sharp', '+go', '+guile', '+io',
        '+lua', '+ocaml', '+php',
        '+python', '+r', '+ruby', '+tcl')

    for b in tested_bindings + untested_bindings:
        if (b in spec):
            depends_on('swig')
```